src/lib/builders/table-of-contents/types.ts

```diff
@@ -0,0 +1,46 @@
+
+export type Heading = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+/**
+ * The ToC type describes which headings should be considered active.
+ */
+export type TOCType = 'lowest' | 'highest' | 'all-active' | 'lowest-parents' | 'highest-parents';
```

Comment:
I'm confused by what this does tbh 🤔

Reply:
Okay, I think I kinda understand, but only partially. This needs to be well documented, maybe adding more info to the JSDoc comment would be stellar as well!

Reply:
Yeah I'll add them to the docs, I wasn't sure if you would even want this option so I didn't expand too much on it for now. Essentially the type defines which headings are considered active, based on the content that is visible. The ones with `-parents` also mark the parent headings of the current active headings as active, and `all-active` marks all headings with visible content as active. Maybe there's a better way to name them?

I was also thinking it would be good to add a `none` option, for those that just want the ToC but don't want to add a bunch of intersection observer APIs to all the elements. What do you think?

Reply:
I think that's a good idea! I love this tbh, it's pretty comprehensive and feature-complete.